### ISSUE TITLE
roachtest: an unexpected error

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#42160]
I wrote a roachtest #42122 that runs TPCH queries against cluster with vectorized off and on. The test reuses the same cluster, so it first loads TPCH data, next runs all TPCH queries with vec off, and then runs a subset of queries with vec on. However, sometimes the second run fails with
```
00:50:31 test.go:284: test failure: 	tpchvec.go:124,test_runner.go:697: /Users/yuzefovich/go/bin/roachprod run yahor-1572655340-01-n3cpu4:1 -- ./workload run tpch --concurrency=1 --db=tpch --max-ops=12 --queries=1,2,3,4,5,6,7,8,11,14,17,18 --vectorize=experimental_always {pgurl:1-3}: exit status 1
```
Looking at the logs, the only suspicious message is
```
242 W191102 00:50:30.483510 191 sql/schema_changer.go:1991  [n3] Error executing schema change: descriptor not found
```
I believe this is caused due to query 15 which creates a view. Does anyone know of any problems in the interaction between schema changes (probably dropping a view) and roachtests? (When query 15 is skipped in the test, then it successfully runs queries on both configs.)

This might be a red herring because I don't see that message in all cases :confused: Anyway, I created a branch (https://github.com/yuzefovich/cockroach/tree/tpch-roachtest-error-repro) with an attempt to make the reproduction shorter.

TL;DR: I expected running two sets of queries to always succeed, but sometimes the second set errors out with a not helpful `exit status 1`, but looking at the logs doesn't tell me what went wrong. It seems like an infrastructure flake.

**Comment by yuzefovich** [comment]
Apparently, adding an automatic retry loop and executing `vec on` queries first doesn't solve this issue in all cases: https://github.com/cockroachdb/cockroach/issues/41108#issuecomment-550118737.

Looking at the statement statistics https://teamcity.cockroachdb.com/repository/download/Cockroach_Nightlies_WorkloadNightly/1575479:id/tpchvec/debug.zip!/debug/nodes/1/crdb_internal.node_statement_statistics.txt, it seems that all the queries are actually executed, but all of them returned 0 rows. In the logs from node 1 (gateway) https://teamcity.cockroachdb.com/repository/download/Cockroach_Nightlies_WorkloadNightly/1575479:id/tpchvec/debug.zip!/debug/nodes/1/logs/cockroach.teamcity-1575479-1573004318-18-n3cpu4-0001.ubuntu.2019-11-06T02_17_30Z.005896.log there are several messages like `context canceled: readerCtx in Inbox stream handler (local reader canceled)`, and I'm not sure what could be the cause here.

**Comment by yuzefovich** [comment]
I now think that the problem was actually me misinterpreting the error (granted, the error doesn't have enough information), and the problem should be fixed by #42292.
